### ISSUE TITLE
Bug: sort-objects doesn't detect dependency from renamed property as default value

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [azat-io/eslint-plugin-perfectionist#491]
### Describe the bug

[`perfectionist/sort-objects`](https://perfectionist.dev/rules/sort-objects) normally detects dependencies from default values - `({ banana, apple = banana })`. But when a previous object is renamed in its destructuring -`({ banana: bananaRenamed, apple = bananaRenamed })`- the rule doesn't seem to detect that.

```plaintext
/Users/josh/repos/repros/index.js
  3:3  error  Expected "apple" to come before "banana"  perfectionist/sort-objects
```

### Code example

export async function runModeSetup({
  banana: bananaRenamed,
  apple = bananaRenamed,
}) {
  console.log(apple, bananaRenamed);
}

### ESLint version

9.22.0

### ESLint Plugin Perfectionist version

4.10.1

### Additional comments

```js
// eslint.config.js
import perfectionist from "eslint-plugin-perfectionist";

export default [
	perfectionist.configs["recommended-natural"],
];
```

Example past issues of dependency detection: #9, #256, #262

### Validations

- [x] Read the [docs](https://perfectionist.dev).
- [x] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @JoshuaKGoldberg,

Thank you for the issue! A fix handling this syntax is on its way, handling syntaxes such as

```ts
let {
  b: bRenamed,
  a = bRenamed,
} = obj;

let [{
  b: bRenamed,
  a = bRenamed,
}] = [obj];

let {
  [b]: bRenamed,
  a = bRenamed,
} = obj;
```

Feel free to mention other unusual syntaxes that we might have missed 🙂

**Comment by azat-io** [comment]
Fixed in a0af5a79abba9c05c757115b5c184ab499efb95e.
Released in [v4.11.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.11.0).
